### PR TITLE
feat(core): replace dotenv with dotenv-expand

### PR DESCRIPTION
**Pull request by xiongemi**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/webpack/package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "copy-webpack-plugin": "^10.2.4",
     "css-loader": "^6.4.0",
     "css-minimizer-webpack-plugin": "^5.0.0",
-    "dotenv": "~16.3.1",
```
Comment (FrozenPandaz) [inline packages/webpack/package.json:41]: I thought we still need dotenv

File: packages/vite/package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
   "dependencies": {
     "@phenomnomnominal/tsquery": "~5.0.1",
     "@swc/helpers": "~0.5.0",
-    "dotenv": "~16.3.1",
```
Comment (FrozenPandaz) [inline packages/vite/package.json:34]: I thought we still need dotenv

File: packages/storybook/package.json [hunk 3]
```diff
@@ -30,7 +30,7 @@
     "migrations": "./migrations.json"
   },
   "dependencies": {
-    "dotenv": "~16.3.1",
```
Comment (FrozenPandaz) [inline packages/storybook/package.json:33]: I thought we still need dotenv

File: packages/webpack/src/executors/webpack/webpack.impl.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import 'dotenv/config';
+import 'dotenv-expand/config';
```
Comment (FrozenPandaz) [inline packages/webpack/src/executors/webpack/webpack.impl.ts:1]: What does this do that the `forked-process-task-runner` doesn't already do?

File: packages/nx/bin/nx.ts [hunk 5]
```diff
@@ -31,7 +32,8 @@ function main() {
   require('nx/src/utils/perf-logging');
 
   performance.mark('loading dotenv files:start');
-  loadDotEnvFiles();
```
Comment (FrozenPandaz) [inline packages/nx/bin/nx.ts:34]: I don't think this has to change? :thinking: This is our own function

File: packages/nx/bin/nx.ts [hunk 6]
```diff
@@ -109,9 +111,10 @@ function main() {
  */
 function loadDotEnvFiles() {
   for (const file of ['.env', '.local.env', '.env.local']) {
-    loadDotEnvFile({
+    var myEnv = loadDotEnvFile({
```
Comment (FrozenPandaz) [inline packages/nx/bin/nx.ts:114]:
```suggestion
    let env = loadDotEnvFile({
```

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
There should be 1 more important usage in `forked-process-task-runner`

File: packages/webpack/package.json (hunk 1 above)
Comment (xiongemi) [reply packages/webpack/package.json]: create a separate pr for this, no loner needed in the package.json

File: packages/vite/package.json (hunk 2 above)
Comment (xiongemi) [reply packages/vite/package.json]: create a separate pr for this, no loner needed in the package.json

File: packages/nx/bin/nx.ts (hunk 5 above)
Comment (xiongemi) [reply packages/nx/bin/nx.ts]: typo, changed back

File: packages/nx/bin/nx.ts (hunk 6 above)
Comment (xiongemi) [reply packages/nx/bin/nx.ts]: changed

File: packages/storybook/package.json (hunk 3 above)
Comment (xiongemi) [reply packages/storybook/package.json]: create a separate pr for this, no loner needed in the package.json

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**